File: src/FashionClothesAndTrends.Domain/Entities/Comment.cs
using FashionClothesAndTrends.Domain.Common;

namespace FashionClothesAndTrends.Domain.Entities;

public class Comment : BaseEntity
{
    public required string Text { get; set; }

    public required string UserId { get; set; }
    public User User { get; set; } = null!;

    public Guid ClothingItemId { get; set; }
    public ClothingItem ClothingItem { get; set; } = null!;

    public ICollection<LikeDislike> LikesDislikes { get; set; } = [];
}

File: src/ClientApp/src/app/shared/models/comment.ts
import { LikeDislike } from "./like-dislike";

export interface Comment {
  id: string;
  text: string;
  username: string;
  userId: string;
  clothingItemId?: string;
  createdAt: Date;
  timeAgo?: string;
  likesCount?: number;
  dislikesCount?: number;
  likesDislikes?: LikeDislike[];
}
